bugwarrior/config.py

```diff
@@ -184,9 +185,27 @@ def get_taskrc_path(conf, main_section):
     )
 
 
-def get_data_path():
-    return os.path.expanduser(os.getenv('TASKDATA', '~/.task'))
+@lru_cache()
+def get_data_path(config, main_section):
+    taskrc = get_taskrc_path(config, main_section)
 
+    # We cannot use the taskw module here because it doesn't really support
+    # the `_` subcommands properly (`rc:` can't be used for them).
+    line_prefix = 'data.location='
+    tw_show = subprocess.Popen(
+        ('task', '_show'), stdout=subprocess.PIPE, env={'TASKRC': taskrc})
+    data_location = subprocess.check_output(
+        ('grep', line_prefix), stdin=tw_show.stdout)
```

Comment:
Probably want to anchor the search pattern with `^` here. I also like passing `-e`, but that's me.